amaranth/hdl/_ir.py

```diff
@@ -601,7 +390,7 @@ def add_signal_name(signal):
 
         return signal_names
 
-    def _assign_names_to_fragments(self, hierarchy=("top",), *, _names=None):
+    def _assign_names_to_fragments(self, hierarchy=("top",), *, top_ports={}, _names=None):
```

Comment:
No mutable objects as default argument.

Reply:
Should probably be called `ports` since otherwise the inner invocation has a lie (the top ports are the same but the value is empty).
```suggestion
    def _assign_names_to_fragments(self, hierarchy=("top",), *, ports={}, _names=None):
```